### PR TITLE
docs: add contributor, maintainer and pr guides

### PR DESCRIPTION
**Pull request by lance**
This commit adds a simple contributors guide, and an in depth set of guidelines
for creating, submitting and updating pull requests. There is also a short set
of suggestions for repository maintainers in a third document.

In an effort to better organize and manage the javascript SDK, I have done this for that repository [here](https://github.com/cloudevents/sdk-javascript/pull/105). There has been [some pushback](https://github.com/cloudevents/sdk-javascript/pull/105#issuecomment-626840358) however because this is not common across all of the SDKs in the organization. I have slightly modified my pull request on the javascript SDK and submitted it here for wider discussion (even though I am not a committer on this project).

The most notable difference between this and what I have suggested for the javascript SDK is that we use the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) fspecification in order to [manage](https://github.com/conventional-changelog/standard-version) the CHANGELOG.md and I thought this might be too big of a change/suggestion for the Go SDK.

Looking forward to the discussion.

Signed-off-by: Lance Ball <lball@redhat.com>

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Contributing to CloudEvents Go SDK
```
Comment (slinkydeveloper) [inline CONTRIBUTING.md:1]: Cloudevents sdk-go

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+Each PR must be signed. Be sure your `git` `user.name` and `user.email` are configured
+then use the `--signoff` flag for your commits.
+
+```console
+git commit --signoff
+```
+
+### Style Guide
+
+Code style for this module is maintained using [`eslint`](https://www.npmjs.com/package/eslint).
```
Comment (slinkydeveloper) [inline CONTRIBUTING.md:30]: leftover :smile: You can replace it with go fmt: https://golang.org/cmd/go/#hdr-Gofmt__reformat__package_sources

File: pr_guidelines.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+need to force push these changes.
+
+```console
+git push -f origin 48-fix-if-present-validation
+```
+
+## Submitting and Updating Your Pull Request
+
+Before submitting a pull request, you should make sure that all of the tests
+successfully pass by running `npm test`.
```
Comment (slinkydeveloper) [inline pr_guidelines.md:102]: leftover from copy paste

File: pr_guidelines.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+* [Congratulations!](#congratulations)
+
+## Getting Started
+
+When creating a pull request, first fork this repository and clone it to your
+local development environment. Then add this repository as the upstream.
+
+```console
+git clone https://github.com/mygithuborg/sdk-go.git
+cd sdk-javascript
```
Comment (slinkydeveloper) [inline pr_guidelines.md:22]: leftover

File: pr_guidelines.md [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+```
+
+## Branches
+
+The first thing you'll need to do is create a branch for your work.
+If you are submitting a pull request that fixes or relates to an existing
+GitHub issue, you can use this in your branch name to keep things organized.
+For example, if you were to create a pull request to fix
+[this error with validation](https://github.com/cloudevents/sdk-go/issues/486)
+you might create a branch named `48-fix-http-agent-error`.
```
Comment (slinkydeveloper) [inline pr_guidelines.md:33]: leftover

File: pr_guidelines.md [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+before your pull request has landed. If there are any commits on `master`
+that conflict with your changes, you may need to update your branch with
+these changes before the pull request can land. Resolve conflicts the same
+way as before.
+
+```console
+git fetch upstream
+git rebase upstream/master
+# fix any potential conflicts
+git push -f origin 48-fix-if-present-validation
```
Comment (slinkydeveloper) [inline pr_guidelines.md:114]: not the same as before

File: maintainer_guidelines.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
+# Maintainer's Guide
```
Comment (slinkydeveloper) [inline maintainer_guidelines.md:1]: I think this guide should be somehow linked from contributing.md

**Comment by lance** [comment]
@slinkydeveloper thanks for the quick review! I've updated the PR with your suggestions.

**Comment by n3wscott** [comment]
LGTM

**Pull request merged**